### PR TITLE
Fuse gen ai and continuous_batching

### PR DESCRIPTION
**Pull request by Wovchena**

### CODE REVIEW

File: src/python/py_generate_pipeline.cpp [hunk 1]
```diff
@@ -534,4 +549,61 @@ PYBIND11_MODULE(py_generate_pipeline, m) {
...
         .def("end", &StreamerBase::end);
+
+    py::class_<GenerationResult>(m, "GenerationResult")
+        .def(py::init<>())
+        .def_readonly("m_request_id", &GenerationResult::m_request_id)
+        .def_property("m_generation_ids",
+            [](GenerationResult &r) -> py::list {
+                py::list res;
+                for (auto s: r.m_generation_ids) {
+                    PyObject* py_s = PyUnicode_DecodeUTF8(s.data(), s.length(), "replace");
```
Comment (mzegla) [inline src/python/py_generate_pipeline.cpp:560]: Can this fail? Shall we do something specific if it does?

File: tests/cpp/scheduler.cpp [hunk 2]
```diff
@@ -17,22 +19,17 @@ void clear_finished_sequences(std::vector<SequenceGroup::Ptr>& requests) {
...
+    configs.at(0).max_num_batched_tokens = 32;
+    configs.at(0).num_kv_blocks = 6;
+    configs.at(0).block_size = 4;
+    configs.at(0).dynamic_split_fuse = false;
+    configs.at(0).max_num_seqs = 5;
+    configs.at(1).max_num_batched_tokens = 32;
+    configs.at(1).num_kv_blocks = 6;
+    configs.at(1).block_size = 4;
+    configs.at(1).dynamic_split_fuse = true;
+    configs.at(1).max_num_seqs = 5;
```
Comment (mzegla) [inline tests/cpp/scheduler.cpp:32]: Why do we need to change that? The previous version was a bit more readable in my opinion.

File: src/python/py_generate_pipeline.cpp (hunk 1 above)
Comment (Wovchena) [reply src/python/py_generate_pipeline.cpp:560]: I believe no. `replace` indicates that if unicode point fails to be decoded it must be replaced with ?. Anyway, I just copied it from python.cpp

File: tests/cpp/scheduler.cpp (hunk 2 above)
Comment (Wovchena) [reply tests/cpp/scheduler.cpp:32]: The current version supports C++17

File: tests/cpp/block_manager.cpp [hunk 3]
```diff
@@ -4,13 +4,13 @@
 
 #include <gtest/gtest.h>
 #include "openvino/runtime/core.hpp"
-#include "continuous_batching_pipeline.hpp"
-#include "sequence_group.hpp"
-#include "scheduler.hpp"
+#include "openvino/genai/continuous_batching_pipeline.hpp"
 #include "openvino/genai/generation_config.hpp"
+#include "../../src/cpp/src/sequence_group.hpp"
```
Comment (iefode) [inline tests/cpp/block_manager.cpp:9]: Please do not  use relative paths?

File: tests/cpp/cache_manager.cpp [hunk 4]
```diff
@@ -4,26 +4,25 @@
 
 #include <gtest/gtest.h>
 #include "openvino/runtime/core.hpp"
-#include "scheduler.hpp"
-#include "device_config.hpp"
-#include "cache_manager.hpp"
+#include "../../src/cpp/src/scheduler.hpp"
```
Comment (iefode) [inline tests/cpp/cache_manager.cpp:7]: Please do not use relative paths

File: tests/cpp/block_manager.cpp (hunk 3 above)
Comment (Wovchena) [reply tests/cpp/block_manager.cpp:9]: I don't see a better way to implement it. `sequence_group.hpp` was private and I preserved that. But the tests need the declaration. I could paste the `sequence_group.hpp`'s content here, but it feels to be even worse.

Comment (iefode) [reply tests/cpp/block_manager.cpp:9]: There is different approach not to use relative paths :) For example, include sources to the cmake tests directly by [`target_include_directories`](https://cmake.org/cmake/help/latest/command/target_include_directories.html)

**Review by iefode — APPROVED** [review]
LGTM

File: tests/cpp/scheduler.cpp (hunk 2 above)
Comment (mzegla) [reply tests/cpp/scheduler.cpp:32]: Do you mean C++ 17 doesn't support this initialization?
I was checking with previous version and switched to cpp 17 and didn't see an error while compiling this part.

Comment (Wovchena) [reply tests/cpp/scheduler.cpp:32]: You must use windows to reproduce. gcc is more forgiving

**Review by mzegla — APPROVED** [review]

**Review by dkalinowski — APPROVED** [review]

File: CMakeLists.txt [hunk 5]
```diff
@@ -23,21 +23,27 @@ project(OpenVINOGenAI
...
     find_package(OpenVINO REQUIRED COMPONENTS Runtime)
 endif()
 
+# check that SDPA to PA transformtion exists
+get_target_property(ov_include_dirs openvino::runtime INTERFACE_INCLUDE_DIRECTORIES)
+find_file(spda_to_pa_header sdpa_to_paged_attention.hpp
+    PATHS ${ov_include_dirs}
+    PATH_SUFFIXES openvino/pass
+    DOC "Path to sdpa_to_paged_attention.hpp header"
+    NO_CACHE REQUIRED NO_DEFAULT_PATH)
```
Comment (ilya-lavrenov) [inline CMakeLists.txt:39]: this code is not required

it was a temporary solution to ensure that OV version to SPDA -> PA transformation is used. Now, we can just set minimal required version to find_package

File: tests/cpp/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,9 @@
+FetchContent_Declare(
```
Comment (ilya-lavrenov) [inline tests/cpp/CMakeLists.txt:1]: let's add copyright header

**Pull request merged**